library/Fission/Challenge/Types.hs

```diff
@@ -0,0 +1,59 @@
+module Fission.Challenge.Types (Challenge(..)) where
+  
+import           Database.Persist.Class
+import           Database.Persist.Types
+import           Database.Persist.Sql
+
+import qualified RIO.ByteString.Lazy as Lazy
+import qualified RIO.Text            as Text
+
+import qualified Network.IPFS.Internal.UTF8 as UTF8
+
+import           Data.Swagger
+import           Servant
+
+import           Fission.Prelude
+
+
+newtype Challenge = Challenge { unChallenge :: Text }
+  deriving          ( Generic )
+  deriving anyclass ( ToSchema
+                    , ToParamSchema
+                    )
+  deriving newtype  ( Eq
+                    , Show
+                    , IsString
+                    , ToHttpApiData
+                    )
+
+
+instance Arbitrary Challenge where
+  arbitrary = Challenge <$> arbitrary
+
+instance ToJSON Challenge where
+  toJSON (Challenge challenge') = String challenge'
+
+instance FromJSON Challenge where
+  parseJSON = withText "Challenge" \txt -> return (Challenge txt)
```

Comment:
👍 

More a pro-tip (no change required), but you can also write this as:

```haskell
withText "Challenge" Challenge
```

Reply:
oh i like it!